### PR TITLE
add space:Storage

### PR DESCRIPTION
**Pull request by bourgeoa**
add space:Storage to headers and  podRoot 'text/turtle' container representation
see issue #1629

### CODE REVIEW

**Review by angelo-v — CHANGES_REQUESTED** [review]
Please add unit tests, at least for the code you added.

File: lib/header.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ function addLinks (res, fileMetadata) {
     addLink(res, 'http://www.w3.org/ns/ldp#DirectContainer', 'type')
   }
   if (fileMetadata.isStorage) {
-    addLink(res, 'http://www.w3.org/ns/pim/space#Storage', 'type')
+    addLink(res, 'http://www.w3.org/ns/pim/space#storage', 'type')
```
Comment (angelo-v) [inline lib/header.js:43]: Why lowercase? http://www.w3.org/ns/pim/space#storage is a property while http://www.w3.org/ns/pim/space#Storage is the class.

this is what CSS sends in the header:

```
link: <http://www.w3.org/ns/pim/space#Storage>; rel="type"
```

Comment (bourgeoa) [reply lib/header.js:43]: Because NSS in profile/card uses space:storage, ESS also and the ontology also.
I think it is an error in CSS.

Comment (jeff-zucker) [reply lib/header.js:43]: 1. The header on </>
```
   link:  <http://www.w3.org/ns/pim/space#Storage>; rel="type".
````
Is the same as the predicate in the turtle of </.meta>
```
   </> rdfs:type  <http://www.w3.org/ns/pim/space#Storage> .
```


2. OTOH, in the profile, the owner/server MAY have a  triple like.
```
   <#me> pim:storage </>.
```
#1 asserts that we are at the Pod root.  #2 asserts that WebId owns the pod.  Some owners may not want to have that fact advertised and there is no requirement that they do.

Comment (kjetilk) [reply lib/header.js:43]: There is a convention to have predicates begin with a lower-case letter, and classes begin with an uppercase letter, and that's what we're seeing here: `space:Storage` is a class, `space:storage` is a predicate, and both are in use. And indeed, as @jeff-zucker is saying, the `Link` header are using classes, so it should be upper case, like @angelo-v objected.

File: lib/ldp-container.js [hunk 2]
```diff
@@ -14,6 +14,14 @@ const path = require('path')
 async function addContainerStats (ldp, reqUri, filename, resourceGraph) {
   const containerStats = await ldp.stat(filename)
   addStats(resourceGraph, reqUri, containerStats, filename)
+  const storage = new URL(reqUri)
+  if (reqUri === storage.origin + '/') {
+    resourceGraph.add(
+      resourceGraph.sym(reqUri),
+      ns.rdf('type'),
+      ns.space('storage')
```
Comment (csarven) [inline lib/ldp-container.js:22]:
```suggestion
      ns.space('Storage')
```

File: lib/header.js [hunk 3]
```diff
@@ -102,6 +106,8 @@ function parseMetadataFromHeader (linkHeader) {
           fileMetadata.isBasicContainer = true
         } else if (parsedLinks[rel] === 'http://www.w3.org/ns/ldp#DirectContainer') {
           fileMetadata.isDirectContainer = true
+        } else if (parsedLinks[rel] === 'http://www.w3.org/ns/pim/space#storage') {
```
Comment (csarven) [inline lib/header.js:109]:
```suggestion
        } else if (parsedLinks[rel] === 'http://www.w3.org/ns/pim/space#Storage') {
```

File: lib/header.js (hunk 1 above)
Comment (csarven) [reply lib/header.js:43]: Confirming that the Solid Protocol requires `http://www.w3.org/ns/pim/space#Storage` (uppercase 'S') for Link rel=type.

```suggestion
    addLink(res, 'http://www.w3.org/ns/pim/space#Storage', 'type')
```

**Review by csarven — CHANGES_REQUESTED** [review]

**Review by kjetilk — APPROVED** [review]
This looks good to me now. However, it has been so long since I was into the details of NSS that it should probably receive more than my review :-)

**Review by angelo-v — APPROVED** [review]

**Pull request merged**